redash/__init__.py

```diff
@@ -68,71 +60,13 @@ def create_redis_connection():
 
 
 setup_logging()
-redis_connection = create_redis_connection()
-
-mail = Mail()
-migrate = Migrate()
-mail.init_mail(settings.all_settings())
-statsd_client = StatsClient(host=settings.STATSD_HOST, port=settings.STATSD_PORT, prefix=settings.STATSD_PREFIX)
-limiter = Limiter(key_func=get_ipaddr, storage_uri=settings.LIMITER_STORAGE)
 
-import_query_runners(settings.QUERY_RUNNERS)
-import_destinations(settings.DESTINATIONS)
+redis_connection = LocalProxy(create_redis_connection)
 
-from redash.version_check import reset_new_version_status
-reset_new_version_status()
+mail = LocalProxy(Mail)
 
+migrate = LocalProxy(Migrate)
 
-class SlugConverter(BaseConverter):
-    def to_python(self, value):
-        # This is ay workaround for when we enable multi-org and some files are being called by the index rule:
-        # for path in settings.STATIC_ASSETS_PATHS:
-        #     full_path = safe_join(path, value)
-        #     if os.path.isfile(full_path):
-        #         raise ValidationError()
-
-        return value
-
-    def to_url(self, value):
-        return value
-
-
-def create_app():
-    from redash import authentication, extensions, handlers
-    from redash.handlers.webpack import configure_webpack
-    from redash.handlers import chrome_logger
-    from redash.models import db, users
-    from redash.metrics.request import provision_app
-    from redash.utils import sentry
-
-    sentry.init()
-
-    app = Flask(__name__,
-                template_folder=settings.STATIC_ASSETS_PATH,
-                static_folder=settings.STATIC_ASSETS_PATH,
-                static_path='/static')
-
-    # Make sure we get the right referral address even behind proxies like nginx.
-    app.wsgi_app = ProxyFix(app.wsgi_app, settings.PROXIES_COUNT)
-    app.url_map.converters['org_slug'] = SlugConverter
-
-    if settings.ENFORCE_HTTPS:
-        SSLify(app, skips=['ping'])
-
-    # configure our database
-    app.config['SQLALCHEMY_DATABASE_URI'] = settings.SQLALCHEMY_DATABASE_URI
-    app.config.update(settings.all_settings())
-
-    provision_app(app)
-    db.init_app(app)
-    migrate.init_app(app, db)
-    mail.init_app(app)
-    authentication.init_app(app)
-    limiter.init_app(app)
-    handlers.init_app(app)
-    configure_webpack(app)
-    extensions.init_extensions(app)
-    chrome_logger.init_app(app)
-    users.init_app(app)
+statsd_client = StatsClient(host=settings.STATSD_HOST, port=settings.STATSD_PORT, prefix=settings.STATSD_PREFIX)
```

Comment:
Those don't need to be proxies since the settings module doesn't import from anywhere else, it's safe to import and use it here.